src/main/resources/TourCode/AjaxActions.xml

```diff
@@ -0,0 +1,141 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc>
+  <web>TourCode</web>
+  <name>AjaxActions</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <parent>TourCode.WebHome</parent>
+  <creator>xwiki:XWiki.Admin</creator>
+  <author>xwiki:XWiki.Admin</author>
+  <customClass/>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <creationDate>1443708279000</creationDate>
+  <date>1445246463000</date>
+  <contentUpdateDate>1445246463000</contentUpdateDate>
+  <version>1.1</version>
+  <title>AjaxActions</title>
+  <defaultTemplate/>
+  <validationScript/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
+  <content>{{velocity}}
+{{html}}
+#**
+* Ajax Actions
+*
+* Author: Mohamed BOUSSAA &lt;mohamed.boussa@xwiki.com&gt; 
+*#
+
+#set($action = $request.getParameter("action"))
+#if ("$!action" != '')
+#if ($action == "editStep" &amp;&amp; "$!request.getParameter('BSTDocRef')" != '')
+  #set($docRef = $request.getParameter("BSTDocRef"))
+  #if ("$!docRef" != '')
+     #set($bootstrapTourDoc = $xwiki.getDocument($docRef))
+     #if ($bootstrapTourDoc)
+        ## Save step
+        #set($title = $request.getParameter("title"))
+        #set($path = $request.getParameter("path"))
+        #set($element = $request.getParameter("element"))
+        #set($backdrop = $request.getParameter("backdrop"))
+        #set($content = $request.getParameter("content"))
+        #set($placement = $request.getParameter("placement"))
+        #set($nbSteps = $request.getParameter("nbSteps"))
+        #set($stepOrder = $request.getParameter("stepOrder"))
+        #set($isNewStep = true)
+        #if ("$!stepOrder" != '') ## Edit
+            #set($stepObj = $bootstrapTourDoc.getObject('TourCode.StepClass','order',$stepOrder))
+            #if($stepObj)
+               #set($isNewStep = false)                
+            #end
+        #end
+        #if ($isNewStep)
+           ## New step
+           #set($stepObj = $bootstrapTourDoc.newObject('TourCode.StepClass'))
+        #end
+        #set($discard = $stepObj.set('path', $path))
+        #set($discard = $stepObj.set('element', $element))
+        #set($discard = $stepObj.set('title', $title))
+        #set($discard = $stepObj.set('placement', $placement))
+        #set($discard = $stepObj.set('backdrop', $backdrop))
+        #set($discard = $stepObj.set('content', $content))
+        #if($isNewStep)
+           #set($discard = $stepObj.set('order', $nbSteps))
+        #end
+        ## Add the javascriptExtention object that contains the Bootstrap tour js code call
```

Comment:
Not fan of that part. I think you should remove it and we'll see how to do it in a cleaner way.